pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -209,15 +209,25 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 	}
 
 	before := pr.Status.GetCondition(duckv1alpha1.ConditionSucceeded)
-	after := resources.GetPipelineConditionStatus(pr.Name, state, c.Logger)
+	after := resources.GetPipelineConditionStatus(pr.Name, pipelineState, c.Logger)
 	pr.Status.SetCondition(after)
 
 	reconciler.EmitEvent(c.Recorder, before, after, pr)
 
+	UpdateTaskRunsStatus(pr, pipelineState)
+
 	c.Logger.Infof("PipelineRun %s status is being set to %s", pr.Name, pr.Status)
 	return nil
 }
 
+func UpdateTaskRunsStatus(pr *v1alpha1.PipelineRun, pipelineState []*resources.PipelineRunTaskRun) {
```

Comment:
I don't think it needs to be exported :angel:

Reply:
oh… because the unit tests runs in separate package (`_test`), gotcha

Reply:
yes, unfortunately the test are in `_test` package so had to

Reply:
just a quick note that we don't need to address now: i find that often when im trying to test a function that is in a private scope, it's a hint that maybe that function makes sense to move into its own package and use publicly (i.e. this function is doing something specific which it makes sense to factor into a public method with a public interface)